packages/flutter_tools/lib/src/protocol_discovery.dart

```diff
@@ -110,7 +138,38 @@ class ProtocolDiscovery {
     if (ipv6) {
       hostUri = hostUri.replace(host: InternetAddress.loopbackIPv6.host);
     }
-
     return hostUri;
   }
 }
+
+/// Throttles a stream by [timeInMilliseconds].
+StreamTransformer<S, S> _throttle<S>({
+  @required int timeInMilliseconds,
+}) {
+  assert(timeInMilliseconds != null);
+
+  S latestLine;
+  int lastExecution;
+  Future<void> throttleFuture;
+
+  return StreamTransformer<S, S>
+    .fromHandlers(
+      handleData: (S value, EventSink<S> sink) {
+        latestLine = value;
+
+        final int currentTime = DateTime.now().millisecondsSinceEpoch;
+        lastExecution ??= currentTime;
+        final int remainingTime = currentTime - lastExecution;
+        final int nextExecutionTime = remainingTime > timeInMilliseconds
+          ? 0
+          : timeInMilliseconds - remainingTime;
+        throttleFuture ??= Future<void>
```

Comment:
This will create a separate Future that completes at the desired time for every log line. Futures that complete 'at the same time' aren't guaranteed to complete in any particular order, so the log messages might get re-arranged. That might not be a problem for your current use-case, but it might confuse someone who wants to try to use this for something else.

I'd suggest trying something simpler. Maybe:
```dart
List<S> latestEvents = <S>[];
...
handleData: (S value, EventSink<S> sink) {
  latestEvents.add(value);
  if (throttleFuture != null) {
    return;
  }
  throttleFuture = Future<void>
    .delayed(Duration(milliseconds: timeInMilliseconds))
    .whenComplete(() {
      sink.addAll(latestEvents);
      latestEvents.clear();
    });
}
```